router/schema.go

```diff
@@ -65,3 +68,48 @@ func (s *staticSchema) Resolvers() Resolvers {
 func (s *staticSchema) Dependencies() []ExecutableSchema {
 	return s.StaticSchemaParams.Dependencies
 }
+
+type Context struct {
+	context.Context
+	graphql.ResolveInfo
```

Comment:
Do we ever need `ResolveInfo`?

If we're not using it at the moment, I think we could:

- Just pass around a `context.Context`
- The day we need `graphql.ResolveInfo` we could just inject it inside the regular context (and then have a `router.ResolveInfo(ctx context.Context) grapnel.ResolveInfo` helper)

No particular reason, just feels more go-y to use context.Context directly rather than a homemade one

Reply:
+1 to just using `context.Context` and storage/retrieval functions. I went maybe _too_ far with that in Bass but it has yet to blow up in my face. :sweat_smile: I like that it just follows the prescribed path with Contexts and avoids any nesting shenanigans as more values need to be stored.

Reply:
Yeah we do actually using `ResolveInfo` in the extension code (in `project.go`).

> The day we need graphql.ResolveInfo we could just inject it inside the regular context (and then have a router.ResolveInfo(ctx context.Context) grapnel.ResolveInfo helper)

I don't think it's that simple here unfortunately. When a graphql query has multiple "branches" (i.e. select multiple fields, use aliases, etc.), they all share the same `context.Context` object even though their `ResolveInfo` object are not the same. So you can't just store `ResolveInfo` at a single key and then retrieve it. I actually had to deal with all this already in the multiplatform PR; my solution there ended up being to use the `Path` in the `ResolveInfo` to keep track of different session context config (like arch). Shockingly, that was actually the *simplest* solution I found. So I don't think we can then also hide `ResolveInfo` in the `Context` and keep using the approach.

I also disagree here for a few other reasons:
1. `context.Context` from the stdlib is an interface, not a struct, so it seems like it was intended that other objects can just implement it. I have seen this done before in popular libraries (though I'd have to go looking around to remember which), so it's not completely unprecedented unless I'm imagining things (always possible)
2. Having a custom struct that implements `context.Context` is nicer than wrapper methods because it's easier to discover what's in the context (it's the fields in the struct) relative to searching all over for any method that could accept a `Context` and return something if it happens to be hidden in your context
3. Multiplatform is the first case where we'll need to pass more information in the context (default arch), but I anticipate in the future we'll add more to that list, so I'd rather just pass that all around in this `router.Context` struct than keep creating more and more wrappers all over

>  avoids any nesting shenanigans as more values need to be stored.

That's a good point, I updated `router.Context` to not use any embedding besides the top level `context.Context` (so it can implement the interface).

Let me know if there's disagreements with all that; I basically just don't see a route to using plain `context.Context` and also doing what we need to do with the multiplatform stuff. I'm sure with more thought we could do it but not sure if worth it.

Reply:
Works for me! Didn't know there was more complexity with using `context.Context`.